### PR TITLE
2: Static typeahead select

### PR DESCRIPTION
**Pull request by davidfloegel**
### Description

This PR introduces a very basic typeahead functionality for the `<Select />` component.
It will automatically filter by the `getOptionLabel` prop (case-insensitive). To make the `<Select />` simply pass in the `searchable` prop.

```
<Select
  searchable
  options={cities}
/>
```

In addition I have also updated some typings and removed shitloads of `any`s :)

### How Has This Been Tested?
- Locally in storybook
- By writing tests

### Types of changes

- [x] New feature (non-breaking change which adds functionality)
- [x] Improvement (non-breaking change which improves functionality or refactors code)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by davidfloegel** [comment]
# GitHub Pages Deployment
Pull Request:
https://sofarsounds.github.io/maestro/pull/219
Branch:
https://sofarsounds.github.io/maestro/branches/component/static-typeahead
Storybook:
https://sofarsounds.github.io/maestro/4266189ebb90435a3b41dfea8d3f90485c6f485e/
Build:
https://sofarsounds.github.io/maestro/4266189ebb90435a3b41dfea8d3f90485c6f485e/dist.zip

**Comment by davidfloegel** [comment]
# GitHub Pages Deployment
Pull Request:
https://sofarsounds.github.io/maestro/pull/219
Branch:
https://sofarsounds.github.io/maestro/branches/component/static-typeahead
Storybook:
https://sofarsounds.github.io/maestro/928bdbadbb3ea98348603f943f226294fac3fb32/
Build:
https://sofarsounds.github.io/maestro/928bdbadbb3ea98348603f943f226294fac3fb32/dist.zip

**Comment by davidfloegel** [comment]
# GitHub Pages Deployment
Pull Request:
https://sofarsounds.github.io/maestro/pull/219
Branch:
https://sofarsounds.github.io/maestro/branches/component/static-typeahead
Storybook:
https://sofarsounds.github.io/maestro/d57582472b7e0887725e78fd7f78d3606c35aefb/
Build:
https://sofarsounds.github.io/maestro/d57582472b7e0887725e78fd7f78d3606c35aefb/dist.zip

**Comment by davidfloegel** [comment]
# GitHub Pages Deployment
Pull Request:
https://sofarsounds.github.io/maestro/pull/219
Branch:
https://sofarsounds.github.io/maestro/branches/component/static-typeahead
Storybook:
https://sofarsounds.github.io/maestro/ba519c5aed4cd414ba776611dc830cfe5b5c6d47/
Build:
https://sofarsounds.github.io/maestro/ba519c5aed4cd414ba776611dc830cfe5b5c6d47/dist.zip

File: src/molecules/Select/Options.tsx [hunk 1]
```diff
@@ -44,6 +44,12 @@ const Options = <T extends {}>({
         width="auto"
       >
         <Menu bordered data-qaid={`${qaId}-menu`}>
+          {options.length === 0 && (
+            <MenuItem disabled data-qaid={`${qaId}-empty`}>
+              No options...
```
Comment (liondedan) [inline src/molecules/Select/Options.tsx]: Could be editable or at least translated?

**Comment by davidfloegel** [comment]
# GitHub Pages Deployment
Pull Request:
https://sofarsounds.github.io/maestro/pull/219
Branch:
https://sofarsounds.github.io/maestro/branches/component/static-typeahead
Storybook:
https://sofarsounds.github.io/maestro/6903ec929021ca7f62dff06ac5ee2a96b9cf36b8/
Build:
https://sofarsounds.github.io/maestro/6903ec929021ca7f62dff06ac5ee2a96b9cf36b8/dist.zip

File: src/molecules/Select/Options.tsx (hunk 1 above)
Comment (davidfloegel) [reply src/molecules/Select/Options.tsx]: could be. but isn't

**Comment by davidfloegel** [comment]
# GitHub Pages Deployment
Pull Request:
https://sofarsounds.github.io/maestro/pull/219
Branch:
https://sofarsounds.github.io/maestro/branches/component/static-typeahead
Storybook:
https://sofarsounds.github.io/maestro/3071cba1e70e0c4411ca19dcba78a8137062a03f/
Build:
https://sofarsounds.github.io/maestro/3071cba1e70e0c4411ca19dcba78a8137062a03f/dist.zip

**Pull request merged**